migrator/migrations/m_192_to_m_193_report_config_migrations/migration_impl.go

```diff
@@ -0,0 +1,210 @@
+		}
+	} else {
+		vulnFilter.CvesSince = &storage.VulnerabilityReportFilters_AllVuln{
+			AllVuln: true,
+		}
+	}
+	newConfig.Filter = &storage.ReportConfiguration_VulnReportFilters{
+		VulnReportFilters: vulnFilter,
+	}
+	return newConfig
+}
+
+func create_notifier(reportConfigProto *storage.ReportConfiguration) *storage.NotifierConfiguration {
+	notifierConfig := &storage.NotifierConfiguration{
+		Ref: &storage.NotifierConfiguration_Id{
+			Id: reportConfigProto.GetEmailConfig().GetNotifierId()},
+		NotifierConfig: &storage.NotifierConfiguration_EmailConfig{
+			EmailConfig: &storage.EmailNotifierConfiguration{
+				MailingLists: reportConfigProto.GetEmailConfig().GetMailingLists(),
+			},
+		}}
+	return notifierConfig
+}
+
+func create_report_snapshot(v1Config *storage.ReportConfiguration, v2Config *storage.ReportConfiguration) *storage.ReportSnapshot {
+
+	// create report snapshot for last successful scheduled report job
+	if v1Config.GetLastSuccessfulRunTime() == nil {
+		return nil
+	}
+	if v1Config.GetLastRunStatus() != nil {
+		return &storage.ReportSnapshot{
+			ReportConfigurationId: v2Config.GetId(),
+			Name:                  v2Config.GetName(),
+			Description:           v2Config.GetDescription(),
+			Type:                  storage.ReportSnapshot_VULNERABILITY,
+			ReportId:              uuid.NewV4().String(),
+			Collection: &storage.CollectionSnapshot{
+				Id: v2Config.GetResourceScope().GetCollectionId(),
+			},
+			Schedule: v2Config.GetSchedule(),
+			ReportStatus: &storage.ReportStatus{
+				RunState:          storage.ReportStatus_DELIVERED,
+				ReportRequestType: storage.ReportStatus_SCHEDULED,
+				CompletedAt:       v1Config.LastSuccessfulRunTime,
+			},
+		}
+	}
+	return nil
+}
+
+func migrate(database *types.Databases) error {
+	db := database.GormDB
+	pgutils.CreateTableFromModel(database.DBCtx, db, updatedSchema.CreateTableReportConfigurationsStmt)
+	db = db.WithContext(database.DBCtx).Table("report_configurations")
+
+	dbSnapshot := database.GormDB
+	pgutils.CreateTableFromModel(database.DBCtx, dbSnapshot, updatedSchema.CreateTableReportSnapshotsStmt)
+	dbSnapshot = dbSnapshot.WithContext(database.DBCtx).Table("report_snapshots")
+
+	dbNotifier := database.GormDB
+	pgutils.CreateTableFromModel(database.DBCtx, dbNotifier, updatedSchema.CreateTableReportConfigurationsStmt.Children[0])
+	dbNotifier = dbNotifier.WithContext(database.DBCtx).Table("report_configurations_notifiers")
+
+	query := db.WithContext(database.DBCtx).Table(updatedSchema.ReportConfigurationsTableName).Select("serialized")
+	rows, err := query.Rows()
+	if err != nil {
+		return errors.Wrapf(err, "failed to iterate table %s", "report_configurations")
+	}
+	defer func() { _ = rows.Close() }()
+	var convertedReportConfigs []*updatedSchema.ReportConfigurations
+	var reportSnapshots []*updatedSchema.ReportSnapshots
+	var reportNotifiers []*updatedSchema.ReportConfigurationsNotifiers
+	for rows.Next() {
+		if rows.Err() != nil {
```

Comment:
not quite.  this should be the first thing outside the loop.  And `utils.Should` is not necessary as the err is actually returned.